sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -71,12 +95,34 @@ case class SortPrefix(child: SortOrder) extends UnaryExpression {
 
   val nullValue = child.child.dataType match {
     case BooleanType | DateType | TimestampType | _: IntegralType =>
-      Long.MinValue
+      if ((child.isAscending && child.nullOrdering == NullFirst) ||
+        (!child.isAscending && child.nullOrdering == NullLast)) {
+          Long.MinValue
```

Comment:
Whether we return a smallest or biggest value for null depending on the combination of the sorting direction and null ordering, because our prefix comparators simply compare 2 values. For DESC, they swap the input parameters (aPrefix, bPrefix) to compare. 

For example, in ASC NULLS LAST case, we want the NULL rows be placed to the last, which means the comparator needs to think the NULL value is the largest. While in DESC NULLS FIRST case, we want the NULL rows be placed to the first, which means the comparator needs to think the NULL value is also the largest.